mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityLinkChunkKafkaHandler.java

```diff
@@ -0,0 +1,334 @@
+        var newRecordId = UUID.randomUUID().toString();
+        var instanceId = bibRecord.getExternalIdsHolder().getInstanceId();
+        var parsedRecord = bibRecord.getParsedRecord();
+        var parsedRecordContent = normalize(parsedRecord.getContent());
+        var fieldsArray = parsedRecordContent.getJsonArray(PARSED_RECORD_FIELDS_KEY);
+
+
+        var updateTargetFieldCodes = extractUpdateTargetFieldCodesForInstance(bibAuthorityLinksUpdate, instanceId);
+        var subfieldChanges = bibAuthorityLinksUpdate.getSubfieldsChanges().stream()
+          .filter(subfieldsChange -> updateTargetFieldCodes.contains(subfieldsChange.getField()))
+          .collect(Collectors.toList());
+
+        var fieldsList = fieldsArray.stream()
+          .map(JsonObject::mapFrom)
+          .map(fieldJson -> fieldJson.getMap().entrySet().stream().findFirst().orElse(null))
+          .filter(Objects::nonNull)
+          .map(field -> {
+            var fieldCode = field.getKey();
+            if (!updateTargetFieldCodes.contains(fieldCode)) {
+              return field;
+            }
+
+            var fieldValueJson = JsonObject.mapFrom(field.getValue());
+            var subfieldsJson = fieldValueJson.getJsonArray(PARSED_RECORD_SUBFIELDS_KEY);
+            if (subfieldsJson == null) {
+              return field;
+            }
+            var subfields = extractSubfields(subfieldsJson);
+
+            var authorityId = extractAuthorityId(subfields);
+            if (authorityId.isEmpty() || !bibAuthorityLinksUpdate.getAuthorityId().equals(authorityId.get())) {
+              return field;
+            }
+
+            var newSubfields = subfieldProcessor.process(fieldCode, subfieldChanges, subfields);
+            log.trace("AuthorityId {}, instanceId {}, field {}, old subfields: {}, new subfields: {}",
+              bibAuthorityLinksUpdate.getAuthorityId(), instanceId, fieldCode, subfieldsJson, newSubfields);
+
+            fieldValueJson.put(PARSED_RECORD_SUBFIELDS_KEY, newSubfields);
+            field.setValue(fieldValueJson);
+            return field;
+          })
+          .collect(Collectors.toList());
+
+        parsedRecordContent.put(PARSED_RECORD_FIELDS_KEY, fieldsList);
+        parsedRecord.setContent(parsedRecordContent.encode());
+        parsedRecord.setFormattedContent(EMPTY);
+        parsedRecord.setId(newRecordId);
+        bibRecord.setId(newRecordId);
+        bibRecord.getRawRecord().setId(newRecordId);
+        bibRecord.setSnapshotId(bibAuthorityLinksUpdate.getJobId());
+      });
+
+      return recordCollection;
+    });
+  }
+
+  private Future<SubfieldProcessor> getSubfieldProcessorForEvent(BibAuthorityLinksUpdate bibAuthorityLinksUpdate) {
+
+    var eventType = bibAuthorityLinksUpdate.getType();
+    switch (eventType) {
+      case DELETE: {
+        log.debug("Precessing DELETE event for authorityId {}", bibAuthorityLinksUpdate.getAuthorityId());
+        return Future.succeededFuture(deleteSubfieldProcessor());
```

Comment:
DELETE type is mainly about delete link and not subfield, so it maybe should be linkProcessor (deleteLinkProcessor, etc.)